instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/TripSessionsBillingTest.kt

```diff
@@ -0,0 +1,210 @@
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isActiveGuidanceSession)
+
+        assertSessionDuration(SESSION_DURATION_SECONDS, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testBillingSessionSentForEmptyActiveGuidanceTrip() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
+        mapboxNavigation.startTripSession()
+        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isActiveGuidanceSession)
+
+        assertSessionDuration(0, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testBillingSessionSentForOneSecondLongFreeDrive() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        delay(TimeUnit.SECONDS.toMillis(SESSION_DURATION_SECONDS))
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+        mapboxNavigation.onDestroy()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isFreeDriveSession)
+
+        assertSessionDuration(SESSION_DURATION_SECONDS, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testBillingSessionSentForEmptyFreeDrive() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+        mapboxNavigation.onDestroy()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isFreeDriveSession)
+
+        assertSessionDuration(0, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testOneBillingSessionSentForPausedAndResumedFreeDrive() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+
+        delay(1000)
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+
+        mapboxNavigation.onDestroy()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isFreeDriveSession)
+
+        assertSessionDuration(0, billingRequests.first().duration)
+    }
```

Comment:
Let's add one more, but for active guidance:
```
        val observer = HttpServiceEventsObserver.register()

        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
        mapboxNavigation.stopTripSessionAndWaitForIdleState()

        delay(1000)

        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
        mapboxNavigation.stopTripSessionAndWaitForIdleState()
        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()

        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)

        val billingRequests = observer.billingRequests()
        assertEquals(1, billingRequests.size)
        assertTrue(billingRequests.first().isActiveGuidanceSession)

        assertSessionDuration(0, billingRequests.first().duration)
```

Reply:
Thanks Vadim, I've added these test cases!